### PR TITLE
Fix Communication between frontend and backend

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
api

### Description
This PR fixes some small issues with the api that came up while letting backend and frontend work together…

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### Related Issue(s)
the corresponding PR in the backend https://github.com/hedgedoc/hedgedoc/pull/1162

### CODE REVIEW

File: public/api/private/config [hunk 1]
```diff
@@ -36,12 +36,13 @@
     "imprint": "https://example.com/imprint"
   },
   "version": {
-    "version": "mock",
-    "sourceCodeUrl": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
-    "issueTrackerUrl": "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
```
Comment (mrdrogdrog) [inline public/api/private/config:41]: Why did you remove these?

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]

File: public/api/private/config (hunk 1 above)
Comment (DerMolly) [reply public/api/private/config:41]: The backend route does not provide those and I think the whole thing can be removed (the version modal needs changes for that), because this was built under the assumption that frontend and backend wouldn't be delivered together. As this is now the case this does not make sense anymore…

Comment (mrdrogdrog) [reply public/api/private/config:41]: So we hardcode the urls now?.. or put them into the version json?

Comment (DerMolly) [reply public/api/private/config:41]: I'm not to sure… I'd suggest removing the two columns and just have the version and the link to the code repo and the issue tracker. Those should the hardcoded then…

Maybe it would be a good idea to add a issue template about GUI issues or something similiar in the hedgedoc/hedgedoc repo which link to an issue in the react-client?

Comment (mrdrogdrog) [reply public/api/private/config:41]: Please don't hard code them. Put them at least into a json file. Or read read them from the package.json.

Comment (DerMolly) [reply public/api/private/config:41]: Seems like a reasonable way of doing things…

Comment (ErikMichelson) [reply public/api/private/config:41]: Reminder that we've got a links.json file...

File: src/api/notes/index.ts [hunk 2]
```diff
@@ -23,11 +23,18 @@ export interface Note {
...
-  })
+  if (process.env.NODE_ENV === 'development') {
+    response = await fetch(getApiUrl() + `/notes/${ noteId }-get`, {
+      ...defaultFetchConfig
+    })
+  } else {
+    response = await fetch(getApiUrl() + `/notes/${ noteId }`, {
+      ...defaultFetchConfig
+    })
+  }
```
Comment (ErikMichelson) [inline src/api/notes/index.ts:37]: Couldn't this be shortened by usage of a ternary operator?

Comment (DerMolly) [reply src/api/notes/index.ts:37]: Something like this?
```
const response = await fetch(getApiUrl() + `/notes/${ noteId }${process.env.NODE_ENV === 'development' ? '-get' : ''}`, {
      ...defaultFetchConfig
    })
```

File: public/api/private/config (hunk 1 above)
Comment (DerMolly) [reply public/api/private/config:41]: I reworked this. Have a look

File: src/api/notes/index.ts (hunk 2 above)
Comment (DerMolly) [reply src/api/notes/index.ts]: Changed to the mentioned solution

Comment (mrdrogdrog) [reply src/api/notes/index.ts]: imho this line is much harder to read now. Could you please merge the getApiURL into the string template and extract the suffix into a var?

Comment (mrdrogdrog) [reply src/api/notes/index.ts]: Also: do you need the spread syntax here?

Comment (ErikMichelson) [reply src/api/notes/index.ts]: The spread syntax isn't needed if the `{ }` around it get removed.

Comment (DerMolly) [reply src/api/notes/index.ts]: I think it's best that this is kept as is. We do it in all fetch calls and it makes it easy to later add extra option if we need them.

**Review by mrdrogdrog — APPROVED** [review]

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]
Cypress tests is failing because the intercept in cypress/support/visit-test-editor.ts#L20 is returning the wrong object.

**Review by mrdrogdrog — APPROVED** [review]

File: public/api/private/notes/old [hunk 3]
```diff
@@ -1,13 +1,18 @@
 {
```
Comment (ErikMichelson) [inline public/api/private/notes/old:1]: Shouldn't this file be named `old-get`?

File: src/utils/is-test-mode.ts [hunk 4]
```diff
@@ -7,3 +7,7 @@
 export const isTestMode = (): boolean => {
   return !!process.env.REACT_APP_TEST_MODE
 }
+
+export const isMockMode = (): boolean => {
```
Comment (ErikMichelson) [inline src/utils/is-test-mode.ts:11]: The name of the file is a bit misleading as it now doesn't contain the isTestMode exclusively.

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

File: public/api/private/notes/old (hunk 3 above)
Comment (mrdrogdrog) [reply public/api/private/notes/old:1]: do we even need this file?

Comment (ErikMichelson) [reply public/api/private/notes/old:1]: Well, the intention of this old note was to test whether the redirect from `/old` to `/n/old` works (cypress test missing btw [but not in this PR]).

Comment (DerMolly) [reply public/api/private/notes/old:1]: Well we should be able to test the redirector without a real backend.

File: src/utils/is-test-mode.ts (hunk 4 above)
Comment (DerMolly) [reply src/utils/is-test-mode.ts]: renamed to `test-modes`

File: public/api/private/notes/old (hunk 3 above)
Comment (DerMolly) [reply public/api/private/notes/old]: fixed

**Comment by ErikMichelson** [comment]
I just realized we'd need a corresponding PR in hedgedoc/react-client-test-page to not break the CI, as the backend doesn't contain the source code and issue tracker URLs in its config anymore.

File: src/utils/test-modes.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+export const testModes = (): boolean => {
```
Comment (ErikMichelson) [inline src/utils/test-modes.ts:7]: This is wrong.

**Review by ErikMichelson — APPROVED** [review]

**Review by ErikMichelson — APPROVED** [review]
